### PR TITLE
Implement immer & some lint fixes

### PR DESCRIPTION
**Pull request by alejobh**
## Summary
* Fixed broken bootstrap README
* Updated stylelint dependencies
* Added and edited useful project commands:
`ts-check`: Checks for typescript errors.
`test`: Updated for ci purposes
`test-interactive`: Same as `test` but keeps on interactive mode
`test-no-coverage`: Same as `test-interactive` but shows no files coverage.
* Removed `import React` from every appearance (not necessary now because @pabloferro updated react version and our eslint rules are mostly up to date thanks to @damfinkel). BUT i had to leave a TODO about a new rule that needs to be added to eslint wolox rules package, but i dont really know how to deal the versioning because its a big breaking change.
* Added **immer** to handle immutability on reducers, with a new typing way using type annotations.

## Known issues
We have to update linters from bootstrap project directly because it's parsing when saving in the wrong way when working on templates.
Also we need to find a way to apply the generated project lint config into the bootstrap itself to code directly in the bootstrap in a safe way without needing to generate a new project and work on there.

## Screenshots
image

image

### CODE REVIEW

File: generators/app/steps/CustomizableSteps/constants.js [hunk 1]
```diff
@@ -69,8 +69,8 @@ module.exports.DEPENDENCIES = [
   'react-hook-form@^6.12.2'
 ];
 
+// TODO: What to do here about dev dependencies when using customizable, why is plop here? shouldn't it be on generators/app/constants.js ?
```
Comment (pabloferro) [inline generators/app/steps/CustomizableSteps/constants.js]: I think we are always using customizable by default for now.
After this remove the comment.

File: generators/app/tasks/fileCreators.js [hunk 2]
```diff
@@ -30,7 +30,10 @@ const getPackageJsonAttributes = (projectName, projectVersion, repoUrl) => ({
     build: 'node ./scripts/build.js',
     // eslint-disable-next-line no-extra-parens
     ...(this.customized && { deploy: 'node ./scripts/deploy.js' }),
-    test: generateRSScript('test', '--env=jsdom'),
+    'ts-check': 'tsc',
```
Comment (pabloferro) [inline generators/app/tasks/fileCreators.js:33]:
```suggestion
    'tsc': 'tsc',
```
Maybe I'd rename the name just for clarity. Not sure 🙃

File: generators/app/templates/templates/component/index.test.tsx.hbs [hunk 3]
```diff
@@ -10,9 +9,19 @@ test('renders correctly without the optionalProp', () => {
...
 test('renders correctly with the optionalProp', () => {
   const { container } = render(<{{name}} someProp={1} someOptionalProp={2} />);
   expect(container.firstChild).toMatchSnapshot();
 });
 
+test('shows optionalProp when sent', () => {
+  const { getByText } = render(<{{name}} someProp={3} someOptionalProp={4} />);
+  expect(getByText(/and 4/i)).toBeInTheDocument();
+});
+
```
Comment (pabloferro) [inline generators/app/templates/templates/component/index.test.tsx.hbs:26]: I'd move the snapshot test to the beginning.

**Review by pabloferro — CHANGES_REQUESTED** [review]
Awesome work 🎉 

I left a few small comments.

File: generators/app/constants.js [hunk 4]
```diff
@@ -113,11 +113,11 @@ module.exports.DEV_DEPENDENCIES = [
...
   '@typescript-eslint/parser@^4.14.2',
-  'stylelint-config-wolox@1.0.7',
-  'stylelint-no-indistinguishable-colors@1.2.1',
-  'stylelint-scss@3.5.4',
-  'stylelint@9.10.1',
+  'stylelint-config-wolox@^1.1.0',
+  'stylelint-scss@^3.18.0',
+  'stylelint@^13.7.0',
   'husky@^4.0.3',
+  'immer@^9.0.1',
```
Comment (pabloferro) [inline generators/app/constants.js]: Move to customizable dependencies (NOT dev dep)

**Review by pabloferro — APPROVED** [review]
Nice work! 🎉

**Comment by pabloferro** [comment]
https://github.com/Wolox/react-bootstrap/pull/163

**Review by damfinkel — APPROVED** [review]

**Pull request merged**